plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java

```diff
@@ -115,11 +115,18 @@ public MemSqlClient(BaseJdbcConfig config, ConnectionFactory connectionFactory,
         this.jsonType = typeManager.getType(new TypeSignature(StandardTypes.JSON));
     }
 
+    @Override
+    public Connection getConnection(ConnectorSession session, JdbcSplit split) throws SQLException
+    {
+        Connection c = super.getConnection(session, split);
+        c.prepareStatement("set collation_connection = 'utf8_bin'").execute();
```

Comment:
This should be done in connection factory.
**For example**, when DELETE is delegated to the database, same settings should be applied.